sdk/containerregistry/Microsoft.Azure.ContainerRegistry/src/Customizations/AcrClientCredentials.cs

```diff
@@ -129,22 +122,36 @@ public AcrClientCredentials(string aadAccessToken, string loginUrl, string tenan
         /// <summary>
         /// Called on initialization of the credentials. This sets forth the type of authorization to be used if necessary.
         /// </summary>
-        public override void InitializeServiceClient<AzureContainerRegistryClient>(ServiceClient<AzureContainerRegistryClient> client)
+        public override void InitializeServiceClient<T>(ServiceClient<T> client)
         {
-            if (_mode == LoginMode.Basic) // Basic Authentication
+            if (client == null)
             {
-                _authHeader = EncodeTo64(_username + ":" + _password);
+                throw new ArgumentNullException(nameof(client));
             }
+
+            // if this is an ACRClient, add the loginUri that this credential was created for
+            if (client is AzureContainerRegistryClient acrClient)
+            {
+                if (acrClient.LoginUri == null)
+                {
+                    acrClient.LoginUri = this._loginUrl;
```

Comment:
because we expect the [client ](https://github.com/AzureCR/azure-sdk-for-net/pull/1/files#diff-0afc02f069edb64af0e8d1f320e1ce69R79) to call this method with itself, we can set the loginUri here. @estebanreyl. This is what I am talking about.